src/Project/Model.cpp

```diff
@@ -0,0 +1,341 @@
+#include <cgltf.h>
+#include <stb_image.h>
+
+#include <Project/Model.hpp>
+
+#include <glm/gtc/type_ptr.hpp>
+
+#include <glad/glad.h>
+#include <GLFW/glfw3.h>
+
+#include <unordered_map>
+#include <filesystem>
+#include <algorithm>
+#include <iterator>
+#include <vector>
+#include <queue>
+
+GLuint64 (*APIENTRY glGetTextureHandleARB)(GLuint texture);
+void (*APIENTRY glMakeTextureHandleResidentARB)(GLuint64 handle);
+
+namespace fs = std::filesystem;
+
+static std::string FindTexturePath(const fs::path& basePath, const cgltf_image* image)
+{
+    std::string texturePath;
+    if (texturePath.empty())
+    {
+        auto newPath = basePath / image->name;
+        if (!newPath.has_extension())
+        {
+            if (std::strcmp(image->mime_type, "image/png") == 0)
+            {
+                newPath.replace_extension("png");
+            }
+            else if (std::strcmp(image->mime_type, "image/jpg") == 0)
+            {
+                newPath.replace_extension("jpg");
+            }
+        }
+        texturePath = newPath.generic_string();
+    }
+    else
+    {
+        texturePath = (basePath / image->uri).generic_string();
+    }
+    return texturePath;
+}
+
+Model::Model(std::string_view file)
+{
+    if (!glGetTextureHandleARB)
```

Comment:
this should be covered by glad out of the box, i dont think checking for null is necessary, we are requesting a gl4.6 context

Reply:
Unfortunately, there are no such functions in `glad.h`, so I had to load them manually.
I think they should be present in `glad.h` if you generate it enabling the required extensions